src/js/controller/instream-adapter.js

```diff
@@ -276,6 +276,14 @@ var InstreamAdapter = function(_controller, _model, _view) {
         }
     };
 
+    this.replacePlaylistItem = function(url) {
```

Comment:
How about `this.setPlaylistItem(item)`? So in the plugin you can use get the item with `api.getPlaylistItem()` make these changes, and this just does `_model.set('playlistItem', item)`.